graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusElement.java

```diff
@@ -105,23 +110,45 @@ public Set<String> getPropertyKeys() {
 
     @Override
     public void removeProperty(String propertyName) {
+        RequestContext context = RequestContext.get();
         Iterator<? extends Property<String>> it = getWrappedElement().properties(propertyName);
         while(it.hasNext()) {
             Property<String> property = it.next();
             property.remove();
+            //fill the map for the entityGuid and remove value
+            if (propertyName.startsWith(INTERNAL_PROPERTY_KEY_PREFIX)) {
```

Comment:
This will collect attrs like __createdBy, __createdAt & other attributes which not denorm attrs. shall we exclude them as well?
or we can even have inclusion list but we will loose future readyness & have to update the inclusion list everytime  we introduce new denorm attribute